barretenberg/cpp/src/barretenberg/stdlib/merkle_tree/membership.hpp

```diff
@@ -41,7 +41,11 @@ field_t<Builder> compute_subtree_root(hash_path<Builder> const& hashes,
         // current iff path_bit If either of these does not hold, then the final computed merkle root will not match
         field_t<Builder> left = field_t<Builder>::conditional_assign(path_bit, hashes[i].first, current);
         field_t<Builder> right = field_t<Builder>::conditional_assign(path_bit, current, hashes[i].second);
-        current = pedersen_hash<Builder>::hash_multiple({ left, right }, 0, is_updating_tree);
+        if (is_updating_tree) {
+            current = pedersen_hash<Builder>::hash({ left, right }, 0);
+        } else {
+            current = pedersen_hash<Builder>::hash_skip_field_validation({ left, right }, 0);
```

Comment:
IIUC this assumes that if someone is not updating the tree, then the input is trusted as we can skip validation checks. I don't know how well that holds -- not a blocker for this PR, but something we probably want to note down

Reply:
This isn't quite correct. I'll ad a clarifying comment.

If we skip field validation checks, then a Prover can provide *two* satisfying witnesses to `hash({left, right})`.

However, if the Prover is not *updating* the tree, they must provide a hash path that produces a *specific* root. As lot as the root was computed correctly (with the field validation checks), a malicious Prover that provides the incorrect satisfying assignment to `hash(left, right)` will not produce a matching root hash.